crates/hotshot/src/lib.rs

```diff
@@ -159,6 +161,26 @@ pub struct SystemContext<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     /// Reference to the internal storage for consensus datum.
     pub storage: Arc<RwLock<I::Storage>>,
 }
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> Clone for SystemContext<TYPES, I> {
```

Comment:
Ouch, caught by the classic `Arc<T::X>` where `T::X` not locally constrained to `Clone` issue.